### PR TITLE
ekncontent: Support "content-type" property on EknQueryObject

### PR DESCRIPTION
**Pull request by smspillaz**
This property can be used to filter a query by content type, for
instance, if you only want to get images or HTML articles.

https://phabricator.endlessm.com/T21112

### CODE REVIEW

File: ekncontent/ekncontent/eknc-query-object.c [hunk 1]
```diff
@@ -51,6 +52,7 @@ struct _EkncQueryObject
   gchar *corrected_terms;
   gchar *stopword_free_terms;
   gchar *literal_query;
+  gchar *content_type;
```
Comment (ptomato) [inline ekncontent/ekncontent/eknc-query-object.c]: nitpick: We're trying to phase out `gchar`, so make this `char`.

File: ekncontent/ekncontent/eknc-query-object.c [hunk 2]
```diff
@@ -456,6 +469,17 @@ eknc_query_object_class_init (EkncQueryObjectClass *klass)
...
       G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
 
+  /**
+   * EkncQueryObject:content-type:
+   *
+   * Content Type to restrict the search to.
+   */
+  eknc_query_object_props[PROP_CONTENT_TYPE] =
+    g_param_spec_string ("content-type", "Content Type",
+      "Content Type to restrict the search to",
```
Comment (ptomato) [inline ekncontent/ekncontent/eknc-query-object.c]: nitpick: `Type` -> `type` (also in the doc comment)

File: ekncontent/ekncontent/eknc-query-object.c [hunk 3]
```diff
@@ -827,6 +866,27 @@ get_corrected_query (EkncQueryObject *self,
   return g_steal_pointer (&query_clause);
 }
 
+/*
```
Comment (ptomato) [inline ekncontent/ekncontent/eknc-query-object.c]: For future reference, static functions don't need this level of detail in the documentation, but it's already here, so best to leave it.

File: ekncontent/ekncontent/eknc-query-object.c [hunk 4]
```diff
@@ -827,6 +866,27 @@ get_corrected_query (EkncQueryObject *self,
...
+ * get_content_type_clause:
+ * @content_type: a string containing the content type
+ *
+ * Retreives a content-type query clause from the content type string
+ *
+ * Returns: (transfer full) (nullable): a #XapianQuery object
+ *
+ */
+static XapianQuery *
+get_content_type_clause (const gchar *content_type)
```
Comment (ptomato) [inline ekncontent/ekncontent/eknc-query-object.c]: `gchar` -> `char`

File: ekncontent/ekncontent/eknc-query-object.c [hunk 5]
```diff
@@ -827,6 +866,27 @@ get_corrected_query (EkncQueryObject *self,
...
+{
+  g_autofree gchar *prefixed = NULL;
```
Comment (ptomato) [inline ekncontent/ekncontent/eknc-query-object.c]: ditto

**Comment by smspillaz** [comment]
Ah yes, of course. I was going to address these comments and then I must have dropped the ball. In any case, I see you have some new comments now, so I'll address them.

File: ekncontent/ekncontent/eknc-query-object.c (hunk 3 above)
Comment (smspillaz) [reply ekncontent/ekncontent/eknc-query-object.c]: Yeah. I was mostly just copying the style I saw in `get_filter_clause`, etc

**Comment by ptomato** [comment]
(Feel free to commit it by the way, once you get `flapjack test eos-knowledge-lib` to pass with gtk-doc enabled.)

**Comment by smspillaz** [comment]
Okay, made that fix. Distcheck passes - I'll allow you to merge when ready.

**Comment by ptomato** [comment]
By "feel free to commit" I meant you could merge it yourself. No worries, anyway we happened to discuss some improvements to this feature this morning, so Martin is going to adopt your commits onto a different pull request. I'll close this one.

**Pull request closed without merging**